src/server/system_services/bucket_server.js

```diff
@@ -1943,17 +1943,25 @@ async function delete_bucket_replication(req) {
     await replication_store.instance().delete_replication_by_id(replication_id);
 }
 
-function validate_replication(req) {
+async function validate_replication(req) {
     const replication_rules = req.rpc_params.replication_policy.rules;
     // num of rules in configuration must be in defined limits
     if (replication_rules.length > config.BUCKET_REPLICATION_MAX_RULES ||
         replication_rules.length < 1) throw new RpcError('INVALID_REPLICATION_POLICY', 'Number of rules is invalid');
 
     const rule_ids = [];
     const pref_by_dst_bucket = {};
+    const src_bucket = req.system.buckets_by_name && req.system.buckets_by_name[req.name];
 
     for (const rule of replication_rules) {
         const { destination_bucket, filter, rule_id } = rule;
+
+        // check for bidirectional replication
+        const repl_rules = filter && await replication_store.instance().get_replication_by_prefix_and_destination_bucket(filter.prefix, src_bucket.id);
+        if (repl_rules != null) {
+            throw new RpcError('INVALID_REPLICATION_POLICY', 'found bidirectional replication');
```

Comment:
maybe more explaining error, I am not sure that `found bidirectional replication` will help the customer/support to understand what needs to be changed.

Reply:
Thanks, I have updated the error msg here.